docs/source/conf.py

```diff
@@ -339,5 +343,37 @@ def _autogenerate():
 _autogenerate()
 
 
+# CUDA target deprecation notice - can be included on any page by including:
+#
+# .. cuda-deprecated::
+
+cuda_deprecation_text = (
+    "The CUDA target built-in to Numba is deprecated, with further development "
+    "moved to the `NVIDIA numba-cuda "
+    "package <https://github.com/NVIDIA/numba-cuda>`_. Please see "
+    ":ref:`cuda-deprecation-status`."
+)
+
+
+class CudaDeprecated(SphinxDirective):
+    def run(self):
+        # Generate a warning admonition to contain the deprecation notice
+        warning = nodes.admonition(classes=["warning"])
+        warning += nodes.title(text="CUDA Target deprecation notice")
+
+        # Parse CUDA deprecation text so that the link and reference get
+        # resolved eventually
+        dummy = nodes.Element()
+        dummy.document = self.state.document
+        lines = StringList(string2lines(cuda_deprecation_text))
+        self.state.nested_parse(lines, 0, dummy)
+
+        # Add the text into the body of the warning
+        warning += dummy.children
```

Comment:
This is an unfamiliar API, I presume it's parsing the text string and making it suitable for inclusion in the warning banner?

Reply:
That's my understanding. The text needs to be prepared so that after a pass over the entire docs, the refs in the text of the notice will be resolved. This is a solution I cobbled together from various bits of documentation and reading the source code for Sphinx that works for all the versions we're using. A simpler solution would have been to use [`parse_text_to_nodes()`](https://www.sphinx-doc.org/en/master/extdev/markupapi.html#parsing-directive-content-as-restructuredtext), but that doesn't exist in the versions we're using / that are available in all places.

Reply:
I see, thanks for figuring this out and for recording an explanation here. It seems to be working ok in practice as the docs render as expected!